hack/macros.py

```diff
@@ -25,24 +59,32 @@ def artifact(repo, file, org="knative"):
         empty this links to googlestorage, otherwise it links via
         the matching release in github.
         """
+
         version = os.environ.get("KNATIVE_VERSION")
+
         if version == None:
-            return 'https://storage.googleapis.com/knative-nightly/{repo}/latest/{file}'.format(
-                    repo=repo,
-                    file=file)
-        else:
-            if version.startswith("v1."):
-                return 'https://github.com/{org}/{repo}/releases/download/knative-{version}/{file}'.format(
-                    repo=repo,
-                    file=file,
-                    version=version,
-                    org=org)
+            return f'https://storage.googleapis.com/knative-nightly/{repo}/latest/{file}'
+        
+        version = version.removeprefix('v')
+
+        try:            
+            v = semver.VersionInfo.parse(version)
+            latest_version = g.latest_release(f'{v.major}.{v.minor}', org, repo)
+
+            if latest_version is None:
+                print_to_stdout(f'repo "{org}/{repo}" has no tags using latest release for file "{file}"')
+                return f'https://github.com/{org}/{repo}/releases/latest/download/{file}'
+            elif version.startswith("1."):
+                return f'https://github.com/{org}/{repo}/releases/download/knative-v{latest_version}/{file}'
             else:
-                return 'https://github.com/{org}/{repo}/releases/download/{version}/{file}'.format(
-                    repo=repo,
-                    file=file,
-                    version=version,
-                    org=org)
+                return f'https://github.com/{org}/{repo}/releases/download/{latest_version}/{file}'
```

Comment:
As of v1.3, we're only supporting docs version of the pattern `knative-v{version}`. Should we remove this block?

Reply:
Good point - dropped